### PR TITLE
Add Tokenizer::tokenize_with_location() API

### PR DESCRIPTION
**Pull request by segoon**
* add TokenWithLocation to track Token location
* implement Tokenizer::tokenize_with_location() -> ``Result<Vec<TokenWithLocation>, ..>``
* refactor line/column tracking to alter current position along with peekable.next() (the old approach with duplicate position tracking was error-prone and contained a bug in multiline word parsing, see the test)
* eliminate String copying for Tokenizer, use ``&'a str`` instead
* minor TokenizerError column fixes, now it shows the missing symbol position and not the expected token start location

### CODE REVIEW

File: src/tokenizer.rs [hunk 1]
```diff
@@ -253,6 +253,31 @@ impl fmt::Display for Whitespace {
     }
 }
 
+/// Location in input string
+#[derive(Debug, PartialEq, Clone)]
```
Comment (Dandandan) [inline src/tokenizer.rs:257]: This could be copy as well, as it is just a struct with `u64,u64`

**Comment by Dandandan** [comment]
This looks really cool @segoon . Could you fix the conflicts?

**Comment by alamb** [comment]
Hi @segoon  -- sorry for the delay in review. I am going to help out now with this repo and we are working to clear the backlog. Is this PR still something you would like to work on to help contribute?

**Comment by alamb** [comment]
I am closing what look like stale PRs in this repo; I apologize in advance if this is a mistake -- please feel free to reopen if you want to keep working on this issue.

**Pull request closed without merging**